lib/module/xkcd.go

```diff
@@ -0,0 +1,70 @@
+package module
+
+import (
+	"fmt"
+	"github.com/davidscholberg/irkbot/lib/configure"
+	"github.com/davidscholberg/irkbot/lib/message"
+	"github.com/nishanths/go-xkcd"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+const apiUrlFmtDefine = "https://relevantxkcd.appspot.com/process?%s"
+
+func Helpxkcd() []string {
+	s := "xkcd <search> - find a xkcd comic relevant to the search term"
+	return []string{s}
+}
+func get(apiURL string) (string, error) {
+	response, err := http.Get(apiURL)
+	if err != nil {
+		return "", err
+	}
+	defer response.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return "", err
+	}
+	bodyString := string(bodyBytes)
+	return bodyString, err
+}
+func fmtString(bodyString string) string {
+	bodyStrings := strings.Split(bodyString, "\n")
+	spacedStrings := strings.Fields(bodyStrings[2])
+	return spacedStrings[0]
+}
+func getXKCD(cfg *configure.Config, in *message.InboundMsg, actions *Actions) {
+	comic := "enter a search term, asshat"
+	//If no search term, gently remind the user to input one
+	if len(in.MsgArgs[1:]) == 0 {
+		actions.Say(comic)
+		return
+	}
+	query := url.Values{}
+	query.Add("action", "xkcd")
+	search := strings.Join(in.MsgArgs[1:], " ")
+	query.Add("query", search)
+	apiUrl := fmt.Sprintf(apiUrlFmtDefine, query.Encode())
+	comicString, comicErr := get(apiUrl)
+	if comicErr != nil {
+		actions.Say("something borked, try again")
```

Comment:
It's good that you alert the channel that there was an error, but you should also print the error itself to stderr so that it can be seen in the program output.

An example of this can be seen here: https://github.com/davidscholberg/irkbot/blob/master/lib/module/slam.go#L53